gdx/src/com/badlogic/gdx/utils/ArrayMap.java

```diff
@@ -398,6 +398,38 @@ public void truncate (int newSize) {
 		size = newSize;
 	}
 
+	@Override
+	public int hashCode () {
+		int h = 0;
+		for (Entry<K, V> entry : entries()) {
+			h += (entry.key != null ? entry.key.hashCode() : 0) * 31;
+			h += (entry.value != null ? entry.value.hashCode() : 0);
+		}
+		return h;
+	}
+
+	@Override
+	public boolean equals (Object obj) {
+		if (obj == this) return true;
+		if (!(obj instanceof ArrayMap)) return false;
+		ArrayMap<K, V> other = (ArrayMap) obj;
+		if (other.size != size) return false;
+		for (Entry<K, V> entry : entries()) {
+			Object otherValue = other.get(entry.key);
+			if (entry.value == null) {
+				if (!other.containsKey(entry.key) || other.get(entry.key) != null) {
+					return false;
+				}
+			} else {
+				if (!entry.value.equals(other.get(entry.key))) {

```

Comment:
Excessive gets. I assume all maps have this so I won't comment about it anymore.